uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/controller/UniProtKBController.java

```diff
@@ -412,4 +424,37 @@ private void setPreviewInfo(UniProtKBSearchRequest searchRequest, boolean previe
             searchRequest.setSize(PREVIEW_SIZE);
         }
     }
+
+    private ResponseEntity<MessageConverterContext<UniProtKBEntry>> redirectToUniSave(String accessionOrId, HttpServletRequest request) {
+        MediaType contentType = getAcceptHeader(request);
+
+        String uniProtPath = ServletUriComponentsBuilder.fromCurrentRequest().build().getPath();
+        String uniSavePath = "";
+        if (uniProtPath != null) {
+            String format = UniProtMediaType.getFileExtension(contentType);
+            String accession = accessionOrId.substring(0, accessionOrId.indexOf('.'));
+            String version = accessionOrId.substring(accessionOrId.indexOf('.') + 1);
+            uniSavePath = uniProtPath.replace("uniprotkb", "unisave");
+            uniSavePath = uniSavePath.substring(0, uniSavePath.lastIndexOf('/') + 1) + accession;
+            uniSavePath = uniSavePath +"?from=" + accessionOrId + "&versions=" + version + "&format=" + format;
+        }
```

Comment:
it would be bit optimal if we use StringBuilder instead of immutable String for uniSavePath